pkg/helm/helm_cli.go

```diff
@@ -66,6 +65,35 @@ func NewHelmCLI(binary string, version Version, cwd string, debug bool, args ...
 	return cli
 }
 
+// NewHelmCLIWithCompatibilityCheck creates a new HelmCLI and checks the compatibility with
+// the currently installed helm CLI. This will exit the program with a fatal log if the helm CLI
+// is not compatible.
+func NewHelmCLIWithCompatibilityCheck(binary string, version Version, cwd string, debug bool, args ...string) *HelmCLI {
+	cli := NewHelmCLI(binary, version, cwd, debug, args...)
+	cli.checkCompatibility()
+	return cli
+}
+
+// checkCompatibility verifies whether the current helm CLI is compatible. The current
+// implementation only supports helm CLI v2. This function will exit the program if the
+// installed helm cli is not compatible.
+func (h *HelmCLI) checkCompatibility() {
+	version, err := h.VersionWithArgs(false, "--client")
+	version = strings.TrimPrefix(version, "Client: ")
+	if err != nil {
+		log.Logger().Warnf("Unable to detect the current helm version due to: %s", err)
+		return
+	}
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		log.Logger().Warnf("Unable to parse the current helm version: %s", version)
+		return
+	}
+	if v.Major > 2 {
+		log.Logger().Fatalf("Your current helm version v%d is not supported. Please downgrade to helm v2.", v.Major)
```

Comment:
I am not a big fan of using `Fatalf`, but I guess w/o further changes that's the best we can do.

One could, however, return an error from `checkCompatibility` or a `bool` and a message `string`.  One can then use `Fatalf` in `NewHelmCLIWithCompatibilityCheck` based on the returned value. The benefit is that ` checkCompatibility`. Up to you. We can also merge as is.

Reply:
I though about about returning an error and let the caller to handle it but if it doesn't treat that error as fatal the execution might continue with an invalid helm version.

Reply:
and the `CreateHelm` doesn't propagate further any error, so we could log it there but keeping  this check encapsulated in the helm client wrapper seems more appropriate to me because we can have in future a different wrapper implementation for helm 3

Reply:
We can merge as it is if you are fine with it.

Reply:
Fair enough. Just wanted to hear your reasoning :-)